tests/test_job.py

```diff
@@ -436,3 +436,274 @@ def update_status(self) -> JobStatus:
     job_builder.submit_pbs()
 
     assert submitted_jobs == ["prepared"]
+
+
+@pytest.mark.parametrize(
+    ("depend_on", "expected"),
+    [
+        pytest.param(
+            "4815162342",
+            "afterok:4815162342",
+            id="basic_job_id",
+        ),
+        pytest.param(
+            "4815162342.datarmor",
+            "afterok:4815162342.datarmor",
+            id="job_id_with_server",
+        ),
+        pytest.param(
+            "afterok:12345",
+            "afterok:12345",
+            id="already_formatted_id",
+        ),
+        pytest.param(
+            ["123", "456", "789"],
+            "afterok:123:456:789",
+            id="already_formatted_id",
+        ),
+    ],
+)
+def test_build_dependency_string_with_string_input(depend_on: str, expected: str) -> None:
+    """Test building dependency string from string inputs."""
+    dep_str = Job._build_dependency_string(depend_on)
+    assert dep_str == expected
+
+
+def test_build_dependency_string_with_single_job() -> None:
+    """Test building dependency string from a single Job instance."""
+    dummy_job = Job(Path("script.py"))
+    dummy_job.job_id = "12345"
+
+    dep_str = Job._build_dependency_string(dummy_job)
+    assert dep_str == "afterok:12345"
+
+
+def test_build_dependency_string_with_job_list() -> None:
+    """Test building dependency string from multiple Job instances."""
+    job1 = Job(Path("script1.py"))
+    job1.job_id = "12345"
+
+    job2 = Job(Path("script2.py"))
+    job2.job_id = "67890"
+
+    job3 = Job(Path("script3.py"))
+    job3.job_id = "11111"
+
+    dep_str = Job._build_dependency_string([job1, job2, job3])
+    assert dep_str == "afterok:12345:67890:11111"
+
+
+def test_build_dependency_string_with_unsubmitted_job_raises() -> None:
+    """Test that using an unsubmitted job raises ValueError."""
+    job = Job(Path("script.py"))
+    job.job_id = None
+    job.name = "my_way"
+
+    with pytest.raises(
+        ValueError, match="Job 'my_way' has not been submitted yet"
+    ):
+        Job._build_dependency_string(job)
```

Comment:
Same comment than before: since `Job` has a `status` property, I'd check that rather than the sole existence of a job ID